### DISCUSSION TITLE
What is the main usecase of graphql-ws? Why use it instead of a regular http transport server?

### DISCUSSION THREAD

**Discussion by capaj** [enisdenjo/graphql-ws#563] [Q&A]
I was looking for some docs on as to why this project exist.
https://the-guild.dev/graphql/ws

has recipes, a basic list of features and a spec section but it never mentions when would you use it?
What is the main usecase? I have always used a regular http transport with graphql in the past and websocket for subscriptions. Why would you want to go with ws transport instead of an http one?

**Comment by enisdenjo** [comment]
Hey there! The graphql-ws/[sse](https://github.com/enisdenjo/graphql-sse)/[http](https://github.com/graphql/graphql-http) suite is about transports that are decided depending on your project's transport requirements.

> Why would you want to go with ws transport instead of an http one?

You don't have to. There is no need to use WS for non-subscription GraphQL.
